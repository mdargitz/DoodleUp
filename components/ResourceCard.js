//title text, image thumbnail, short description from parent

import React from 'react';
import { View, TouchableOpacity, Text, Image, Linking } from 'react-native';

export default function ResourceCard(props) {
  return (
    <View>
      <TouchableOpacity
        onPress={() => Linking.openURL('https://www.sophia.org/tutorials/introduction-to-gesture-drawing')}
        style={styles.touchStyle}
      >
        <Image
          source={{ uri: props.image }}
          style={{ width: 200, height: 200 }}
        />
        <Text style={styles.textStyle}>{props.title}</Text>
      </TouchableOpacity>
    </View>

  );
}

const styles = {
  touchStyle : {
    backgroundColor: '#3B3B58',
    justifyContent: 'center',
    alignItems: 'center',
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.8,
    shadowRadius: 2,
    padding: 30,
    margin: 20
  },
  textStyle : {
    color: '#fff',
    fontSize: 20,
    padding: 10,
    textAlign: 'center'
  }
};