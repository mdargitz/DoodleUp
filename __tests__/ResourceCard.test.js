import React from 'react';
import ResourceCard from '../components/ResourceCard';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(<ResourceCard />).toJSON();
  expect(tree).toMatchSnapshot();
});