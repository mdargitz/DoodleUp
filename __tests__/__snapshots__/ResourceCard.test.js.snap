// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#3B3B58",
        "justifyContent": "center",
        "margin": 20,
        "opacity": 1,
        "padding": 30,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.8,
        "shadowRadius": 2,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Image
      source={
        Object {
          "uri": undefined,
        }
      }
      style={
        Object {
          "height": 200,
          "width": 200,
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#fff",
          "fontSize": 20,
          "padding": 10,
          "textAlign": "center",
        }
      }
    />
  </View>
</View>
`;
